src/deepsparse/transformers/utils/eval_helpers.py

```diff
@@ -0,0 +1,94 @@
+from transformers import AutoTokenizer
+from datasets import load_dataset
+import numpy
+
+
+def process_concatenated_datasets(dataset_name, model_path, max_sequence_length, kwargs):
+    if dataset_name == "wikitext2":
+        eos = kwargs.get("eos", "\n\n")
+        bos = kwargs.get("bos", "")
+
+        raw_dataset = load_dataset("wikitext", "wikitext-2-raw-v1", split="test")
+        raw_text = raw_dataset["text"]
+    elif dataset_name == "c4":
+        eos = kwargs.get("eos", "<|endoftext|>")
+        bos = kwargs.get("bos", "")
+        raw_samples = kwargs.get("raw_samples", None)
+        data_file = kwargs.get("data_file", 0)
+        if data_file is not None:
+            raw_dataset = load_dataset(
+                "allenai/c4",
+                "allenai--c4",
+                data_files={"validation": f"en/c4-validation.{data_file:05d}-of-00008.json.gz"},
+                split="validation",
+            )
+        else:
+            raw_dataset = load_dataset(
+                "allenai/c4",
+                "allenai--c4",
+                split="validation",
+            )
+        if raw_samples is not None:
+            raw_dataset = raw_dataset[:raw_samples]
+        raw_text = raw_dataset["text"]
```

Comment:
This function uses a ton of hardcoded names and expects a certain pattern for file names. 

- Please add a test to ensure the filename assumptions hold and the different if/else conditions are valid for whatever inputs we expect
- Could we define constants at the top and use them here?

Reply:
Fixed